packages/nodes-base/nodes/Oracle/Sql/helpers/interfaces.ts

```diff
@@ -0,0 +1,125 @@
+import type { DateTime } from 'luxon';
+import type { IDataObject, INodeExecutionData } from 'n8n-workflow';
+import type * as oracleDBTypes from 'oracledb';
+
+export type QueryMode = 'single' | 'transaction' | 'independently';
+
+export type ObjectQueryValue = Extract<oracleDBTypes.BindParameters, Record<string, unknown>>;
+
+// For execute
+export type QueryValue =
+	| ObjectQueryValue // named binds in object form
+	| oracleDBTypes.BindParameters; // positional binds in array form
+
+// A query string along with its bind values.
+export type QueryWithValues = {
+	query: string;
+	values?: QueryValue; // For execute
+	executeManyValues?: QueryValue[]; // for executeMany
+};
+
+export type WhereClause = { column: string; condition: string; value: any };
+export type SortRule = { column: string; direction: string };
+export type ColumnInfo = {
+	columnName: string;
+	dataType: string;
+	isNullable: boolean;
+	udtName?: string;
+	columnDefault?: string | null;
+	isGenerated?: 'ALWAYS' | 'NEVER';
+	identityGeneration?: 'ALWAYS' | 'NEVER';
+	maxSize: number;
+};
+
+export type QueriesRunner = (
+	queries: QueryWithValues[],
+	items: INodeExecutionData[],
+	options: IDataObject,
+) => Promise<INodeExecutionData[]>;
+
+export type OracleDBNodeOptions = {
+	nodeVersion?: number;
+	operation?: string;
+
+	// Connection options
+	poolPingInterval?: number;
+	poolPingTimeout?: number;
+	stmtCacheSize?: number;
+	poolMax?: number;
+	poolMin?: number;
+	poolIncrement?: number;
+
+	// Execute options
+	autoCommit?: boolean;
+	bindDefs?: oracleDBTypes.BindDefinition[];
+	batchErrors?: boolean;
+	fetchArraySize?: number;
+	keepInStmtCache?: boolean;
+	maxRows?: number;
+	prefetchRows?: number;
+
+	// n8n options
+	largeNumbersOutputAsString?: boolean; // bigInt
+	outputColumns?: string[];
+	stmtBatching?: QueryMode;
+	executeManyOptions?: oracleDBTypes.ExecuteManyOptions;
+};
+
+export type OracleDBNodeCredentials = {
+	accessToken?: any;
+	connectionString: string | undefined;
+	connectionClass?: string;
+	connectTimeout?: number;
+	configDir?: string;
```

Comment:
Correct me if I'm wrong, but I think `accessToken` and `configDir` aren't used anywhere
```suggestion
	connectionString: string | undefined;
	connectionClass?: string;
	connectTimeout?: number;
```